src/Config/Validator/Deploy/ElasticSuiteConfiguration.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\MagentoCloud\Config\Validator\Deploy;
+
+use Magento\MagentoCloud\Config\Stage\DeployInterface;
+use Magento\MagentoCloud\Config\Validator;
+use Magento\MagentoCloud\Config\ValidatorInterface;
+use Magento\MagentoCloud\Process\Deploy\InstallUpdate\ConfigUpdate\SearchEngine\ElasticSuite;
+
+/**
+ * Validates different aspects of ElasticSuite's configuration.
+ */
+class ElasticSuiteConfiguration implements ValidatorInterface
+{
+    /**
+     * @var ElasticSuite
+     */
+    private $elasticSuite;
+
+    /**
+     * @var Validator\ResultFactory
+     */
+    private $resultFactory;
+
+    /**
+     * @var DeployInterface
+     */
+    private $config;
+
+    /**
+     * @param ElasticSuite $elasticSuite
+     * @param Validator\ResultFactory $resultFactory
+     * @param DeployInterface $config
+     */
+    public function __construct(
+        ElasticSuite $elasticSuite,
+        Validator\ResultFactory $resultFactory,
+        DeployInterface $config
+    ) {
+        $this->elasticSuite = $elasticSuite;
+        $this->resultFactory = $resultFactory;
+        $this->config = $config;
+    }
+
+    /**
+     * @return Validator\ResultInterface
+     */
+    public function validate(): Validator\ResultInterface
+    {
+        if (!$this->elasticSuite->isInstalled()) {
+            return $this->resultFactory->success();
+        }
+
+        if (!$this->elasticSuite->isAvailable()) {
+            return $this->resultFactory->error('ElasticSuite is installed without available ElasticSearch service.');
```

Comment:
how do you know that if ElasticSuite is not available then the problem exactly in the ES ;) ?
This text of error should be generated in that class where you validate ES, or use validation of ES here.

Reply:
Because of `isAvailable` method

```
$this->elasticSearch->isInstalled() && $this->isInstalled();
```